Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Community List View.swift

```diff
@@ -0,0 +1,398 @@
+                    proxy.scrollTo(viewId, anchor: .center)
+                    
+                    // Play nice tappy taps
+                    let impact = UIImpactFeedbackGenerator(style: .rigid)
+                    impact.impactOccurred()
+                }
+            }
+        }
+        return Rectangle().fill(Color.clear)
+    }
+}
+
+///
+// Sidebar Label Views
+///
+struct SectionIndexText: View {
+    let label: String
+    var body: some View {
+        Text(label).font(.system(size:12)).bold()
+    }
+}
+
+struct SectionIndexImage: View {
+    let image: String
+    var body: some View {
+        Image(systemName: image).resizable()
+            .frame(width: 8, height: 8)
+    }
+}
+
+///
+// Preview-only functions to populate the model
+// Not sure on what best practices are here
+// If these are useful these could be moved to be used externally
+///
+let fakeCommunityPrefixes: [String] =
+// Generate A-Z
+Array(65...90).map({
+    var asciiStr = ""
+    asciiStr.append(Character(UnicodeScalar($0)!))
+    return asciiStr
+}) +
+Array(97...122).map({
+    var asciiStr = ""
+    asciiStr.append(Character(UnicodeScalar($0)!))
+    return asciiStr
+}) +
+// Generate A bunch of randomm ASCII to make sure sorting works
+Array(33...95).map({
+    var asciiStr = ""
+    asciiStr.append(Character(UnicodeScalar($0)!))
+    return asciiStr
+})
+
+func generateFakeCommunity(id: Int, namePrefix: String) -> APICommunity {
+    return APICommunity(id: id, name: "\(namePrefix) Fake Community \(id)", title: "\(namePrefix) Fake Community \(id) Title", description: "This is a fake community (#\(id))", published: Date.now, updated: nil, removed: false, deleted: false, nsfw: false, actorId: URL(string: "lemmy\(id).google.com")!, local: false, icon: nil, banner: nil, hidden: false, postingRestrictedToMods: false, instanceId: 0)
+}
+    
+func generateFakeFavoritedCommunity(id: Int, namePrefix: String) -> FavoriteCommunity {
+    return FavoriteCommunity(forAccountID: 0, community: generateFakeCommunity(id: id, namePrefix: namePrefix))
+}
+
+struct CommunityListViewPreview: PreviewProvider {
+    static let favoritesTracker: FavoriteCommunitiesTracker = FavoriteCommunitiesTracker(favoriteCommunities: [
+        generateFakeFavoritedCommunity(id: 0, namePrefix: fakeCommunityPrefixes[0]),
+        generateFakeFavoritedCommunity(id: 20, namePrefix: fakeCommunityPrefixes[20]),
+        generateFakeFavoritedCommunity(id: 10, namePrefix: fakeCommunityPrefixes[10])
+    ])
+    
+    static var previews: some View {
+        CommunityListView(
+            account: SavedAccount(id: 0, instanceLink: URL(string: "lemmy.com")!, accessToken: "abcdefg", username: "Test Account"),
+            testCommunities: fakeCommunityPrefixes.enumerated().map({
+                (index, element) in
+                generateFakeCommunity(id: index, namePrefix: element)
+            })
+        ).environmentObject(favoritesTracker)
+    }
+}
+    
```

Comment:
Maybe we should move other components into separate files? I won't hold the PR up for this, it's just a thought.